vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -0,0 +1,30 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OrderHandler
+
+      def self.process(occupancy)
+        return unless occupancy.orderable?
+
+        order = Order.create!(
```

Comment:
You should validate that the `state` is "purchased", `ordered_at` is set. I believe we'll need to call `order.validate_order!` and `order.purchase!` to do those transitions.

We should check `Order#cart_valid?` to see if there's anything else we need to test. I think we've already do most of the `has_valid_payment?` checks at the account selection level, so I'm not that worried about it (e.g. credit cards and POs are facility specific, so they shouldn't be displayed in the context of another facility on the kiosk). And that the `User#accounts_for_product` that we do in the account selection rule does indeed check that you have a valid price group when you select an account.

I think I'm probably ok if we raise an invalid state transition error all the way up in this case. It might be worth writing a spec that stubs out `Order#cart_valid?` to return false and see what happens. We might want to capture it and return something to the doors service (what does it do right now if it gets a 500 from us?)